### ISSUE TITLE
Datagrid: Apply filter is automatically occurring when using date picker on the filter row

### ISSUE THREAD

**Issue by claudenbach** [infor-design/enterprise#5215]
Describe the bug
Apply filter is automatically occurring when picking a date using the date picker on the filter row.

1. Go to https://master-enterprise.demo.design.infor.com/components/datagrid/test-filter-datepicker.html
2. Click on the date picker for the Date column in the filter row
3. Pick a date
4. When the datepicker is closed is performs an applyFilter

Expected behavior
Similar to other filterable columns, the applyFilter is not automatically invoked.

Version
- ids-enterprise: v4.51.0-dev

Additional context
Issue being reported by custFlintHillsResources via https://jira.lawson.com/browse/LMCLIENT-35806

**Comment by tmcconechy** [comment]
When i was thinking about this i thought that the datepicker should autoselect as it is like a dropdown. But maybe we can make it not run automatically when `filterWhenTyping=false`

It seems like extra effort to hit enter after you select

**Comment by claudenbach** [comment]
Keying off the filterWhenTyping will work for us.  Will also need this for filtering a dropdown field as well.  Are there other column types were applyFiltering is performed automatically?  Clients want to control when filtering occurs.

**Comment by tmcconechy** [comment]
The dropdown will be a problem as there isnt any place to hit enter when it closes. This feels really awkward to me.
So they want when the close it to have to hit enter? But if they click it again it will open then they have to reselect the value, close it and hit enter.

Maybe we can introduce a column setting? But i think we might have to kick this to design and get a button added?

Upping the estimate

**Comment by claudenbach** [comment]
Agree.  I'll chat to Phillip and set something up if needed.

**Comment by claudenbach** [comment]
Let's leave the dropdown column and codeblock column as is in regard to auto filtering.

**Comment by tmcconechy** [comment]
Ok so just change the datepicker? Got it

**Comment by pwpatton** [comment]
@tmcconechy I think this also applies to dropdown and multi select dropdown fields. To get passed this we've overridden some keys and behavior in lmcleint.

This is required for us as each time the filter runs it could be expensive. So the idea is that the user sets all their criteria and runs the filter 

- Enter key runs the the filter when dropdown and pickers are not displayed.
- Space bar opens the dropdown, datepicker, multiselect
- enter and space bar will select a value in "picker" and close the picker: except for multi select dropdown
- multi select dropdown will require the enter to close and space bar will select.

**Comment by tmcconechy** [comment]
This seems like an incorrect keyboard combination. For example space bar to open the pickers.
I think what would be better is to keep the normal keyboard functionality and then add a button on the filter row you can click to run the filter?

**Comment by pwpatton** [comment]
that would be great! not having an affordance has been a bummer for us.

We've also added a hot key and if that was displayed in a tooltip over the button that would really help with discovery.

**Comment by tmcconechy** [comment]
Yeah the only problem technically is the layout so would have to figure out a way to add the button? Maybe it needs a column to be added at the end.

I think thats the best solution so lets see if we can get design help.

cc @katieinfor

**Comment by tmcconechy** [comment]
@pwpatton what do you think about something like this where a dedicated button would enable on the toolbar https://www.figma.com/file/o1Ct7fr9uRtkEUJnVesZXm/UXD-2944?node-id=7%3A2842 depending on the filter state (this is from design)
